odl/solvers/functional/default_functionals.py

```diff
@@ -1258,17 +1259,18 @@ def _call(self, x):
         # Lazy import to improve `import odl` time
         import scipy.special
 
-        if self.prior is None:
-            tmp = self.domain.element(
-                -1.0 * (np.log(1 - x))).inner(self.domain.one())
-        else:
-            tmp = self.domain.element(-scipy.special.xlogy(
-                self.prior, 1 - x)).inner(self.domain.one())
-        if np.isnan(tmp):
+        with np.errstate(invalid='ignore'):
+            if self.prior is None:
+                res = -(np.log(1 - x)).inner(self.domain.one())
+            else:
+                xlogy = scipy.special.xlogy(self.prior, 1 - x)
+                res = -self.domain.element(xlogy).inner(self.domain.one())
+
+        if not np.isfinite(res):
             # In this case, some element was larger than or equal to one
             return np.inf
```

Comment:
This has to be wrong, right? The gradient is a space element.

Reply:
I see what you mean, issue of display :-) The folded part is all the way from `KullbackLeibler.gradient` to `KullbackLeiblerCrossEntropy._call`.
So this code is part of `KullbackLeiblerCrossEntropy._call`.

Reply:
Ok, then I'm good!